packages/core-commands/src/admin-navigation-commands.js

```diff
@@ -1,9 +1,76 @@
 /**
  * WordPress dependencies
  */
-import { useCommand } from '@wordpress/commands';
+import { useCommand, useCommandLoader } from '@wordpress/commands';
 import { __ } from '@wordpress/i18n';
 import { plus } from '@wordpress/icons';
+import { getPath } from '@wordpress/url';
+import { store as coreStore } from '@wordpress/core-data';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useMemo, useCallback } from '@wordpress/element';
+
+function useAddNewPageCommand() {
+	const isSiteEditor = getPath( window.location.href )?.includes(
+		'site-editor.php'
+	);
+	const { userCanCreatePages } = useSelect( ( select ) => {
+		const { canUser } = select( coreStore );
+		return {
+			userCanCreatePages: canUser( 'create', {
```

Comment:
This doesn't exist yet. Could be an overkill, but if the current user can't create pages I think it's okay not to expose that command.